### ISSUE TITLE
Consistent handling of optional functionality

### ISSUE THREAD

**Issue by Cito** [PyGreSQL/PyGreSQL#12]
PyGreSQL can currently be compiled with various options that would enable or disable parts of the functionality like large object support or default parameters. These options can be added as user options during setup and are passed to the compiler as preprocessor definitions.

This makes the documentation and code overly complicated. Instead, the complete functionality should be enabled by default. There should only be one "compatibility version" setting that would make it possible to not include functionality that is only supported in newer libpq versions, in order to be able to provide binaries compatible with older libpq versions.

Also, the behavior for options that are not enabled should be consistent: Either these methods/attributes should simply not be provided (using them will raise AttributeErrors) or the should be always available but using them should raise NotSupportedError when they require a newer libpq version. Currently the behavior is mostly the former, but not very consistent. The latter has the advantage that the behavior will be the same when an old version of libpq is installed and used at runtime, and applications need only one type of check to safeguard against old libpq versions.

**Comment by Cito** [comment]
Note: Btw, the documentation of the names of the preprocessor definitions is currently not in line with the code. They used to be called `NO_` that needed to be defined to disable, but now have been renamed to positive form and need to be defined to enable.

**Comment by Cito** [comment]
Note 2: The current implementation with boolean build switches like `direct-access` in setup.py which are enabled by default is a bit silly anway, since there is no way to set such switches to false via the command line. Currently you can only pass `--direct-access`, but not `--direct-access=false` or `direct-access=true`  or `--no-direct-access`. So something needs to be changed there anyway.

**Comment by Cito** [comment]
In short, the setup.py should only pass one option, the compatible libpq version. The C code should make preprocessor checks using the maximum of that verstion and the PG_VERSION_NUM included via pgconfig.h and runtime checks using PQlibVersion() if using newer functionality.

**Comment by Cito** [comment]
The issues mentioned in Note 1 and Note 2 have been fixed in the master branch. All switches now have negative variants, like `--direct-access` and `--no-direct-access`. By default everything is enabled and you get warnings if features are not available, or errors if they are explicitly required with these switches. This is more similar to what we have been doing in the past than the idea of using a "libpq compatibility version". I have created separate issues #36 and #37 for other problems mentioned here and will close this one now.
